src/GraphQL/Builders/ConnectionBuilder_Typed.cs

```diff
@@ -0,0 +1,291 @@
+        /// </summary>
+        /// <typeparam name="TNodeType">The graph type of the connection's node.</typeparam>
+        public static ConnectionBuilder<TSourceType, TReturnType> Create<TNodeType>(string name = "default")
+            where TNodeType : IGraphType => Create<TNodeType, EdgeType<TNodeType>>(name);
+
+        /// <summary>
+        /// Returns a builder for new connection field for the specified node and edge type.
+        /// The connection type is <see cref="ConnectionType{TNodeType, TEdgeType}">ConnectionType</see>&lt;<typeparamref name="TNodeType"/>, <typeparamref name="TEdgeType"/>&gt;
+        /// </summary>
+        /// <typeparam name="TNodeType">The graph type of the connection's node.</typeparam>
+        /// <typeparam name="TEdgeType">The graph type of the connection's edge. Must derive from <see cref="EdgeType{TNodeType}">EdgeType</see>&lt;<typeparamref name="TNodeType"/>&gt;.</typeparam>
+        public static ConnectionBuilder<TSourceType, TReturnType> Create<TNodeType, TEdgeType>(string name = "default")
+            where TNodeType : IGraphType
+            where TEdgeType : EdgeType<TNodeType>
+            => Create<TNodeType, TEdgeType, ConnectionType<TNodeType, TEdgeType>>(name);
+
+        /// <summary>
+        /// Returns a builder for new connection field for the specified node, edge and connection type.
+        /// </summary>
+        /// <typeparam name="TNodeType">The graph type of the connection's node.</typeparam>
+        /// <typeparam name="TEdgeType">The graph type of the connection's edge. Must derive from <see cref="EdgeType{TNodeType}">EdgeType</see>&lt;<typeparamref name="TNodeType"/>&gt;.</typeparam>
+        /// <typeparam name="TConnectionType">The graph type of the connection. Must derive from <see cref="ConnectionType{TNodeType, TEdgeType}">ConnectionType</see>&lt;<typeparamref name="TNodeType"/>, <typeparamref name="TEdgeType"/>&gt;.</typeparam>
+        public static ConnectionBuilder<TSourceType, TReturnType> Create<TNodeType, TEdgeType, TConnectionType>(string name = "default")
+            where TNodeType : IGraphType
+            where TEdgeType : EdgeType<TNodeType>
+            where TConnectionType : ConnectionType<TNodeType, TEdgeType>
+        {
+            var fieldType = new FieldType
+            {
+                Name = name,
+                Type = typeof(TConnectionType),
+                Arguments = new QueryArguments(),
+            };
+            fieldType.Arguments.Add(new QueryArgument(typeof(StringGraphType))
+            {
+                Name = "after",
+                Description = "Only look at connected edges with cursors greater than the value of `after`.",
+            });
+            fieldType.Arguments.Add(new QueryArgument(typeof(IntGraphType))
+            {
+                Name = "first",
+                Description = "Specifies the number of edges to return starting from `after` or the first entry if `after` is not specified.",
```

Comment:
```suggestion
                Description = "Specifies the maximum number of edges to return, starting after the cursor specified by `after`, or the first number of edges if `after` is not specified.",
```
I rewrote this a bit more as it was inaccurate (does not start at 'after').

Reply:
If we are okay making changes to the description field in between versions, we can make the same change to the untyped builder. I’d say it’s fine for a new minor version (4.x) but not a new revision version (4.2.x).

Reply:
OK